### ISSUE TITLE
Google+ API Shutdown

### ISSUE THREAD

**Issue by nysky1** [mstade/passport-google-oauth2#13]
Google+ API is shutting down on March 9, 2019.  The progressive shutdown begins as early as January!

https://developers.google.com/+/api-shutdown

**Comment by julianlam** [comment]
So far as I know, there are no plans to discontinue [plain OAuth 2.0 authentication](https://developers.google.com/identity/choose-auth), so passport-google-oauth2 *should* continue working, but I would definitely appreciate some confirmation on that.

**Comment by rwky** [comment]
@julianlam is correct, the plan OAuth2 is sticking around (and is now what users should switch to instead of plus).  So everything is fine.

**Comment by CatalinaMoisuc** [comment]
See the documentation added here: https://github.com/jaredhanson/passport-google-oauth2/pull/51/files#diff-04c6e90faac2675aa89e2176d2eec7d8R102 on how to make make it use the userinfo endpoint instead of the old google+ one.

**Comment by rwky** [comment]
I stand corrected the login uses the google oauth2 apis the profile uses the old plus api

https://github.com/mstade/passport-google-oauth2/blob/master/lib/oauth2.js#L91

**Comment by rwky** [comment]
This fork resolves the problem https://github.com/passport-next/passport-google-oauth2

**Comment by julianlam** [comment]
Isn't changing the `userRoute` enough to become compliant with the Google+ API shutdown?

**Comment by nysky1** [comment]
Did you complete a pull request @rwky ...

**Comment by rwky** [comment]
@nysky1 nope the repo is a fork not intended for merging back into this one. You can use the fork it has packages published in npm.

**Comment by minotogna** [comment]
anyone has an update on how to use the new auth api with this? Why having different node modules ?

**Comment by julianlam** [comment]
@minotogna FWIW here's the change I made:

https://github.com/julianlam/nodebb-plugin-sso-google/commit/8b8392c128743df391382cb9ee4415bc426c492c

It seems to work with existing implementations (id and secret), although I have no freaking clue how to generate a new client id and secret for a brand new install ever since Google changed their backend layout. Anyone got tips?

**Comment by minotogna** [comment]
Thank you @julianlam ! I will give it a try!

**Comment by jeremiepape** [comment]
I've changed those lines to work without Google+ : 
https://github.com/mstade/passport-google-oauth2/blob/59253f8449d1200d07aa61e4ae7205f75e175159/lib/oauth2.js#L45-L46 with :
```
  options.authorizationURL = options.authorizationURL || 'https://accounts.google.com/o/oauth2/v2/auth';
  options.tokenURL = options.tokenURL || 'https://www.googleapis.com/oauth2/v4/token';
```

https://github.com/mstade/passport-google-oauth2/blob/59253f8449d1200d07aa61e4ae7205f75e175159/lib/oauth2.js#L91 with :
`  this._oauth2.get('https://www.googleapis.com/oauth2/v3/userinfo', accessToken, function (err, body, res) {`

In my Google Strategy I use "email" not "profile, so I've changed for : 

```
  passport.use('google', new GoogleStrategy({

      clientID: config.google.clientID,
      clientSecret: config.google.clientSecret,
      callbackURL: config.google.callbackURL,
      passReqToCallback : true

    }, (req, accessToken, refreshToken, email, done) => {
```
Finally, in your routes file you can request like this : 
`passport.authenticate('google', { scope: ['email'] })`
or like this if you need profile (do not forget to add it in your GoogleStrategy callback)
`passport.authenticate('google', { scope: ['email', 'profile'] })`

**Comment by kogal** [comment]
Hey guys! are you gonna fix this?

**Comment by Guisch** [comment]
Hey guys

I did this PR based on @jeremiepape advice. I also updated some other field in profile to maximise compatibility between the current one and the new one.
Don't forget that google is shutting down https://www.googleapis.com/plus/v1/people/me on march 8 2019 (tomorrow).
Reviews is welcomed (for example @mstade)

**Comment by mstade** [comment]
Resolved in #15 – many thanks for the assist @Guisch!
